tests/upgrades-images-unsigned/upgrade_test.go

```diff
@@ -10,13 +10,21 @@ import (
 var _ = Describe("cOS Upgrade tests - Images unsigned", func() {
 	var s *sut.SUT
 
+	BeforeSuite(func() {
+		sut.SnapshotVagrant()
+	})
+
+	AfterSuite(func() {
+		sut.SnapshotVagrantDelete()
+	})
+
 	BeforeEach(func() {
 		s = sut.NewSUT()
 		s.EventuallyConnects(360)
 	})
 
 	AfterEach(func() {
-		s.Reset()
+		sut.ResetWithVagrant()
```

Comment:
I was also wondering it would it be better to simply remove the `AfterEach` and append its logic to `BeforeEach`, probably it adds an extra step on the first one, but at least it does not run a `Reset` after a failure. On failure I would expect to simpley error out quick and do nothing (so it keeps the failed environment locally and it also exits quick on GHA).

Reply:
I can plug that only in case of success we should reset the vm back, good suggestion!